bcda/cli.go

```diff
@@ -67,18 +120,28 @@ func importCCLF8(fileMetadata cclfFileMetadata) error {
 	)
 
 	sc := bufio.NewScanner(file)
+	count := 0
+	validator := cclfFileValidator["CCLF8"]
+
 	for sc.Scan() {
 		b := sc.Bytes()
-		if len(bytes.TrimSpace(b)) > 0 {
+		bytelength := len(bytes.TrimSpace(b))
+		if bytelength > 0 && bytelength <= validator.maxRecordLength {
+			count++
+			// currently only errors if there are more records than we expect.
+			if count > validator.totalRecordCount {
+				return errors.New("maximum record count reached: did not match cclf0 validation")
+			}
 			fmt.Printf("\nMBI: %s\n", b[mbiStart:mbiEnd])
 			fmt.Printf("HICN: %s\n", b[hicnStart:hicnEnd])
+		} else {
+			return errors.New("incorrect record length: did not match cclf0 validation")
```

Comment:
Same as above.  Can we log as much as possible here? Filename being validated? Expected count vs actual count? Anything else?

Reply:
same response above